### ISSUE TITLE
New ModBus version from Lambda

### ISSUE THREAD

**Issue by bs0001** [RalfWinter/lambda-heatpump-modbus-tcp-HA#15]
Lambda has now a updated document of ModBus from 18.09.2024 ([link](https://lambda-wp.at/wp-content/uploads/2024/09/Modbus-Protokoll-und-Beschreibung.pdf))

Probably now some more features are available for remote functions of EUxxL and read of heatpump data.
If possible it would be nice if you'd integrate them in the future.

**Comment by bs0001** [comment]
Hi,

I did some checks about nomenklatures and values. I write it in German.
Based on above ModBus infos some changes might be senseful:

HA = Home Assistant
WP = Wärmepumpe


**eu08l_Hp1_Energy_source_inlet_temperature** (aktueller Wert in HA: 100,1°C)
eine Dezimalstelle zu viel. Korrekt lt. wäre lt. WP 10,0°C
--> Korrektur bei der Übertragung

**eu08l_Hp1_Energy_source_outlet_temperature** (aktueller Wert in HA: -3.276,8°C)
Da kann ich aktuell nichts dazu sagen, da mir in der WP der Wert nicht angezeigt wird.
Evtl. ist aber auch hier eine Dezimalstelle zu viel. Ggf. checken.

**eu08l_hp1_request-type** (aktueller Wert HA: 0)
eu08l_hp1_request-type_txt (aktuell Wert HA: Nicht Verfügbar)
unsere WP läuft gerade, "domestic hot water" müsste hier angezeigt werden.
"domestic hot water" steht aber in Entität eu08l_hp1_operating_state_txt
--> Korrektur der ModBus Zuordnung

**eu08l_Hp1_Operating_State** (aktueller Wert HA: 2)
WP zeigt im WP-Modul folgendes an: Automatik / Brauchwasser / Regelbetrieb / Laufzeit

**eu08l_hp1_operating_state_txt** (aktueller Wert HA: domestic hot water)
WP zeigt im WP-Modul folgendes an: Automatik / Brauchwasser / Regelbetrieb / Laufzeit
--> Demnach müsste hier vermutlich "DHW" bzw. dessen deutsche Bezeichnung drinnen stehen



**eu08l_boil1_operating_state_txt** (aktueller Wert HA: prompt-dhw)
--> Wäre evtl. schön, wenn die adäquaten deutschen Begriffe hier angezeigt werden


**eu08l_Boil1_Error_Number** (aktueller Wert HA: 0)
--> Wäre evtl. im Zuge gleicher Nomenklatur sinnvoll, die Entität so zu benennen: eu08l_Boil1_Error_State

**eu08l_Boil1_error_state_txt**
--> Ist aktuell nicht verfügbar. Wäre evtl. sinnvoll, sie aufzunehmen. Lt. WP müsste als Text "No Error" drinnen stehen. Evtl. generell Bezeichnungen in deutsch machen.



**eu08l_Buff1_Error_Number** (aktueller Wert HA: 0)
--> Wäre evtl. im Zuge gleicher Nomenklatur sinnvoll, die Entität so zu benennen: eu08l_Buff1_Error_State

**eu08l_Buff1_error_state_txt**
--> Ist aktuell nicht verfügbar. Wäre evtl. sinnvoll, sie aufzunehmen. Lt. WP müsste als Text "No Error" drinnen stehen. Evtl. generell Bezeichnungen in deutsch machen.



**eu08l_Sol1_Error_Number** (aktueller Wert HA: 0)
--> Wäre evtl. im Zuge gleicher Nomenklatur sinnvoll, die Entität so zu benennen: eu08l_Sol1_Error_State

**eu08l_Sol1_error_state_txt**
--> Ist aktuell nicht verfügbar. Wäre evtl. sinnvoll, sie aufzunehmen. Lt. WP müsste als Text "No Error" drinnen stehen. Evtl. generell Bezeichnungen in deutsch machen.



**eu08l_Hc1_Error_Number** (aktueller Wert HA: 0)
--> Wäre evtl. im Zuge gleicher Nomenklatur sinnvoll, die Entität so zu benennen: eu08l_Hc1_Error_State

**eu08l_Hc1_error_state_txt**
--> Ist aktuell nicht verfügbar. Wäre evtl. sinnvoll, sie aufzunehmen. Lt. WP müsste als Text "No Error" drinnen stehen. Evtl. generell Bezeichnungen in deutsch machen.



It would be great if you can do the adjustments in upcoming days/weeks.

BR
Bernd

**Comment by RalfWinter** [comment]
Let me comment on your points:

**EU13L_Hp1_Energy_source_inlet_temperature**: corrected, scale: 0.01
**EU13L_Hp1_Energy_source_outlet_temperature**: corrected, scale: 0.01
**EU13L_Hp1_Request-Type_txt**: changed all text conversions from "int" to "int(0)" in order to specify the default value when no update for the derived value has been received yet. This will fix all *_txt entities.
**All *_Error_Number" entities**: "Error Number" is the definition in Lambda's Modbus document for this value - no change
**German value names**: this is the international version of HA-integration. Lambda has started to market their heatpumps in other countries than Germany and Austria. Therefore I leave this in English. Feel free to change your local copy of this yaml-file according to your needs.

**Comment by bs0001** [comment]
Hi Ralf,

perfect, thanks a lot!

Meanwhile I'd probably match the English expressions with suitable German one.

With EU13L_Hp1_Request-Type_txt I currently have an issue.
By showing the entity I got the following message:
`Diese Entität („sensor.eu08l_hp1_request_type_txt“) hat keine eindeutige ID, daher können die Einstellungen nicht über die UI verwaltet werden. Schaue in der [Dokumentation](https://www.home-assistant.io/faq/unique_id) nach für mehr Details.`

I will first update your corrections and will see if the issue has been solved then.

Quite nice to get more and more familiar with the heatpump.

BR
Bernd

**Comment by bs0001** [comment]
One hint:
At the end of the configuration file there's a missing "]".
Correction to:
`          {{ ['Off', 'Manual', 'Automatik', 'Auto-Heating', 'Auto-Cooling', 'Frost', 'Summer', 'Floor-dry'][states('sensor.eu13l_hc2_operating_mode')|int(0)] }}`

The topic with the Request-Type now also works 👌

**Comment by RalfWinter** [comment]
> At the end of the configuration file there's a missing "]".

Yepp, saw it, corrected in the source.

**Comment by RalfWinter** [comment]
> With EU13L_Hp1_Request-Type_txt I currently have an issue. By showing the entity I got the following message: `Diese Entität („sensor.eu08l_hp1_request_type_txt“) hat keine eindeutige ID, daher können die Einstellungen nicht über die UI verwaltet werden. Schaue in der [Dokumentation](https://www.home-assistant.io/faq/unique_id) nach für mehr Details.`
> 
> I will first update your corrections and will see if the issue has been solved then.

Yes, Bernd, please first apply the changes. In my HA I can display the entity without error:
![grafik](https://github.com/user-attachments/assets/71132283-ab23-4bd6-959c-a64ca01689c8)

**Comment by bs0001** [comment]
The following topics are currently not available via ModBus, isn't it?

- Zirkulationspumpenstatus
- Zirkulationstemperatur
- Setting of operation time frames for heatpump, boiler and buffer

I'll collect some ideas and approvements and send it from time to time to Lambda for hopeful future integration.

**Comment by RalfWinter** [comment]
Well, I also miss the 5x05 Modbus values:
`Setpoint flow line temp. RW INT16 [0.1°C] Setpoint temperature flow line (min = 15.0°C; max = 65.0°C)`

They worked initially but after a SW update only -300 is displayed.

**Comment by bs0001** [comment]
yes, correct. Value is -300 and not shown. I will also put this infos in my message to Lambda.
My Lamdba is still running since 09/2024. Therefore I have no older values in charts where I can see when the value wasn't available.

**Comment by bs0001** [comment]
In the configuration-file there's a small misstake in line 540:
`         - name: eu08l_Hc2_Romm_device_temperature`

Correct will be:
`         - name: eu08l_Hc2_Room_device_temperature`

I changed this on my local version already.

**Comment by RalfWinter** [comment]
Fixed typo.

**Comment by RalphDiesinger** [comment]
Hallo,
es gibt nun wieder eine neue Version des Modbus-Dokuments. In den Registern ist mir keine Neuerung aufgefallen. Nur auf Seite 8 unter Punkt 5.1 Kommunikations Einstellungen findet sich nun ein anderer Text. Hier bin ich bzgl. des ACHTUNG aufmerksam geworden. Hat das eine Auswirkung auf die Implementierung in Homeassistant so wie es derzeit umgesetzt ist ??. Link zur neuen Beschreibung: https://lambda-wp.at/wp-content/uploads/2024/11/Modbus-Protokoll-und-Beschreibung.pdf

Hier die Stelle:
Unit ID ist 1  
• Kommunikation erfolgt über Port 502  
• Es können bis zu 16 Kommunikationskanäle (16 Master) bedient werden. 
• Die Server IP Adresse wird in der Steuerung auf der Seite „Netzwerkeinstellungen“ angezeigt. 
• ACHTUNG!!!: 
Die Verbindung darf nicht bei jeder Modbusanforderung aufgebaut und wieder geschlossen 
werden. Ansonsten kann es zu schweren Störungen kommen. 

Grüße Ralph

**Comment by tcm8180** [comment]
The [link](https://lambda-wp.at/wp-content/uploads/2024/10/Modbus-Protokoll-und-Beschreibung.pdf) on the [Lambda download page](https://lambda-wp.at/downloads-2/) is broken. 

I found this one:
https://lambda-wp.at/wp-content/uploads/2025/02/Modbus-Protokoll-und-Beschreibung-1.pdf
but no link to it. Maybe it's unreleased new version.
